share/spice/detect_lang/detect_lang.js

```diff
@@ -1,149 +1,153 @@
-function ddg_spice_detect_lang (api_result) {
+(function(env) {
     "use strict";
+    env.ddg_spice_detect_lang = function (api_result) {
 
-    // Check for any errors.
-    if(!api_result || !api_result.data || !api_result.data.detections || api_result.data.detections.length === 0) {
-        return Spice.failed('detect_lang');
-    }
+        // Check for any errors.
+        if(!api_result || !api_result.data || !api_result.data.detections || api_result.data.detections.length === 0) {
+            return Spice.failed('detect_lang');
+        }
 
-    var query = "";
-    $("script").each(function() {
-        var matched, result;
-        matched = $(this).attr("src");
-        if(matched) {
-            result = matched.match(/\/js\/spice\/detect_lang\/([^\/]+)/);
-            if(result) {
-                query = decodeURIComponent(result[1]);
+        var query = "";
+        $("script").each(function() {
+            var matched, result;
+            matched = $(this).attr("src");
+            if(matched) {
+                result = matched.match(/\/js\/spice\/detect_lang\/([^\/]+)/);
+                if(result) {
+                    query = decodeURIComponent(result[1]);
+                }
             }
-        }
-    });
+        });
 
-    api_result.data.detections.sort(function(a, b) {
-        if(a.confidence > b.confidence) {
-            return -1;
-        } else if(a.confidence < b.confidence) {
-            return 1;
-        }
-        return 0;
-    });
+        api_result.data.detections.sort(function(a, b) {
```

Comment:
According to a comment on #988 this is okay because there are only two
results.

On Sunday 12 June 2016, gaulrobe <notifications@github.com> wrote:

> In share/spice/detect_lang/detect_lang.js
> <https://github.com/duckduckgo/zeroclickinfo-spice/pull/2806#discussion_r66712388>
> :
>
> >
> > -    api_result.data.detections.sort(function(a, b) {
> > -        if(a.confidence > b.confidence) {
> > -            return -1;
> > -        } else if(a.confidence < b.confidence) {
> > -            return 1;
> > -        }
> > -        return 0;
> > -    });
> > +        api_result.data.detections.sort(function(a, b) {
>
> @sahildua2305 <https://github.com/sahildua2305> this should be ported to
> use sort_fields per
> http://docs.duckduckhack.com/frontend-reference/display-reference.html#sortfields-object-optional
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-spice/pull/2806/files/e93f5a62af89c35e55e37f6ae01bf996f3331dc8#r66712388>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AE9xBU6u372kKm3SfDIzZPGgv8ZMPypCks5qKzE-gaJpZM4Izllg>
> .
>


-- 
Regards
*Sahil Dua*
Netaji Subhas Institute of Technology, New Delhi
Connect on LinkedIn <http://in.linkedin.com/in/sahildua2305>
*www.sahildua.com <http://www.sahildua.com>*